test/server/listener_manager_impl_test.cc

```diff
@@ -141,22 +141,31 @@ class ListenerManagerImplWithRealFiltersTest : public ListenerManagerImplTest {
                                                                                     context);
             }));
     socket_ = std::make_unique<NiceMock<Network::MockConnectionSocket>>();
-    address_.reset(new Network::Address::Ipv4Instance("127.0.0.1", 1234));
+    local_address_.reset(new Network::Address::Ipv4Instance("127.0.0.1", 1234));
+    remote_address_.reset(new Network::Address::Ipv4Instance("127.0.0.1", 1234));
   }
 
   const Network::FilterChain*
-  findFilterChain(uint16_t destination_port, bool expect_destination_port_match,
+  findFilterChain(uint16_t destination_port, const uint8_t times, const std::string& source_address,
```

Comment:
In order to retain existing style, could you:
1. Leave `expect_destination_port_match` as is,
2. Add `expect_application_protocol_match` which increases `times` if true,
3. Add `source_address` at the end of the list (to keep it consistent with the matching order),
4. Move `remote_address_` code at the end of the function (again, to keep it consistent with the matching order).
5. Use `Times()` instead of unbounded `WillRepeatedly()`, to verify number of times the remote address is accessed.

Reply:
Done with small interpretation. I guess you meant `expect_source_type_match` not `expect_application_protocol_match`.
In any case the code looks better now. Hope it's (almost) final. Thanks.